pinot-tools/src/main/java/org/apache/pinot/tools/PinotDataAndQueryAnonymizer.java

```diff
@@ -0,0 +1,1287 @@
+      String literalValue = literalAstNode.getValueAsString();
+      String derivedValue = (String)getGeneratedValueForOrigValue(columnName, literalValue);
+      if (literalAstNode instanceof StringLiteralAstNode) {
+        // quote string literals
+        sb.append("\"").append(derivedValue).append("\"");
+      } else {
+        // numeric literals
+        sb.append(derivedValue);
+      }
+    }
+
+    private void loadGlobalDictionariesAndColumnMapping() throws Exception {
+      // load column name mapping
+      File mappingFile = new File(_outputDir + "/" + COLUMN_MAPPING_FILE_KEY);
+      InputStream inputStream = new FileInputStream(mappingFile);
+      BufferedReader reader = new BufferedReader(new InputStreamReader(inputStream));
+      String line;
+      while ((line = reader.readLine()) != null) {
+        String[] names = line.split(COLUMN_MAPPING_SEPARATOR);
+        _origToDerivedColumnsMap.put(names[0], names[1]);
+      }
+
+      // load global dictionaries
+      for (String column : _globalDictionaryColumns) {
+        Map<Object, Object> origToDerived = new HashMap<>();
+        _origToDerivedValueMap.put(column, origToDerived);
+        File dictionaryFile = new File(_outputDir + "/" + column + DICT_FILE_EXTENSION);
+        inputStream = new FileInputStream(dictionaryFile);
+        reader = new BufferedReader(new InputStreamReader(inputStream));
+        String line1;
+        while ((line1 = reader.readLine()) != null) {
+          String line2 = reader.readLine();
+          origToDerived.put(line1, line2);
+        }
+      }
+    }
+
+    private Object getGeneratedValueForOrigValue(String column, Object origValue) {
+      // we use this method for only those columns during query generation that are
+      // actually occurring in predicates
+      // Two kinds of columns are supported in predicates by query generator
+      // (1) columns that are participating in filter (and thus we built global
+      //     dictionary for them containing 1-1 mapping between original and
+      //     generated value
+      // (2) columns that we retained the data for as is.
+      if (_columnsNotAnonymized.contains(column)) {
+        // if we retained the value, simply return the original value
+        return origValue;
+      } else if (_globalDictionaryColumns.contains(column)) {
+        // if we built global dictionary, then get the generated value from global dictionary
+        Map<Object, Object> origToDerived = _origToDerivedValueMap.get(column);
+        Preconditions.checkState(origToDerived != null);
+        return origToDerived.get(origValue);
+      } else {
+        // based on the current implementation, we should not come here since
+        // we preprocess the queries to identify the filter columns and the columns
+        // we retain the data for
+        throw new IllegalStateException("Encountered an invalid filter column: " + column);
```

Comment:
We need to handle the case in which we may have filters like "WHERE foo > 5" but 5 is never there in any segment. This can be a TODO

Reply:
Done